controllers/secret_controller.go

```diff
@@ -149,6 +161,10 @@ func (r *SecretReconciler) Reconcile(ctx context.Context,
 			result.Requeue, reconcileErr)
 	}()
 
+	if request.NamespacedName.Name == secrets.TLSSecret {
+		return reconcile.Result{}, r.reconcileTLSSecret(ctx)
+	}
```

Comment:
By adding this to the function, this `Reconcile` function is now no longer has a single purpose, but is reconciling different secrets in different ways.
Please move the rest of this function into a new function `reconcileUserDataSecret` so the separation between these two purposes is clear.